### PR TITLE
JAMES-2713 DMV Routes Integration test Purge and Delete API

### PR DESCRIPTION
**Pull request by hoangdat**

### CODE REVIEW

**Comment by hoangdat** [comment]
test this please

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java [hunk 1]
```diff
@@ -598,6 +609,136 @@ public void vaultExportShouldResponseIdempotentSideEffect() throws Exception {
...
+
+        purgeVault();
+
+        String fileLocation = exportAndGetFileLocationFromLastMail(EXPORT_ALL_HOMER_MESSAGES_TO_BART, bartAccessToken);
+        try (ZipAssert zipAssert = assertThatZip(new FileInputStream(fileLocation))) {
+            zipAssert.hasEntriesSize(1)
+                .allSatisfies(entry -> entry.hasName(messageIdOfHomer + ".eml"));
+        }
+    }
+
```
Comment (Arsnael) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: Maybe add a purge test with multiple emails? and one with expired emails and not expired?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java [hunk 2]
```diff
@@ -602,6 +609,67 @@ public void vaultExportShouldResponseIdempotentSideEffect() throws Exception {
         }
     }
 
+    @Test
+    public void vaultPurgeShouldDeleteExpiredMessageThenExportWithNoEntry() throws Exception {
```
Comment (trantienduchn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: `vaultPurgeShouldMakeExportProduceEmptyZipWhenAllMessagesAreExpired` take it if you think it's better

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java [hunk 3]
```diff
@@ -602,6 +609,67 @@ public void vaultExportShouldResponseIdempotentSideEffect() throws Exception {
...
+        Instant now = Instant.now();
+        when(clock.instant())
+            .thenReturn(now, now.plus(366, ChronoUnit.DAYS));
```
Comment (trantienduchn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: extract `now.plus(366, ChronoUnit.DAYS)` to an `Instant oneDayAfterExpiration` helps me understand it better.

And you can replace 366 by `RetiontionConfiguration.getRetentionPeriod().plusDays(1)`

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java [hunk 4]
```diff
@@ -602,6 +609,67 @@ public void vaultExportShouldResponseIdempotentSideEffect() throws Exception {
...
+        purgeVault();
+
+        String fileLocation = exportAndGetFileLocationFromLastMail(EXPORT_ALL_HOMER_MESSAGES_TO_BART, bartAccessToken);
+        try (ZipAssert zipAssert = assertThatZip(new FileInputStream(fileLocation))) {
+            zipAssert.hasNoEntry();
+        }
+    }
+
+    @Test
+    public void vaultPurgeShouldNotDeleteNotExpiredMessageThenExportWithAnEntry() throws Exception {
```
Comment (trantienduchn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: `vaultPurgeShouldMakeExportProduceAZipWithAllNonExpiredMessages` take it if you think it's better

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java (hunk 1 above)
Comment (hoangdat) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: added

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java [hunk 5]
```diff
@@ -630,15 +636,18 @@ public void vaultPurgeShouldDeleteExpiredMessageThenExportWithNoEntry() throws E
     }
 
     @Test
-    public void vaultPurgeShouldNotDeleteNotExpiredMessageThenExportWithAnEntry() throws Exception {
-        Instant now = Instant.now();
+    public void vaultPurgeShouldMakeExportProduceAZipWhenOneMessageAreNotExpired() throws Exception {
```
Comment (Arsnael) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: `vaultPurgeShouldMakeExportProduceAZipWhenOneMessageAreNotExpired` -> `vaultPurgeShouldMakeExportProduceAZipWhenOneMessageIsNotExpired`

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java [hunk 6]
```diff
@@ -610,13 +612,17 @@ public void vaultExportShouldResponseIdempotentSideEffect() throws Exception {
...
+            .thenReturn(NOW, NOW, NOW);
 
         bartSendMessageToHomer();
-        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 1);
+        bartSendMessageToHomer();
+        bartSendMessageToHomer();
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 3);
+
+        when(clock.instant())
+            .thenReturn(NOW, ONE_DAY_AFTER_ONE_YEAR_EXPIRATION);
```
Comment (Arsnael) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: Does this make all 3 messages expired? When I look at the code overall I doubt it... but I could be wrong !

**Pull request merged**